python/tskit/trees.py

```diff
@@ -1600,6 +1600,36 @@ def newick(self, precision=14, root=None, node_labels=None):
             return self.__build_newick(root, precision, node_labels) + ";"
         return s
 
+    def as_dict_of_dicts(self):
```

Comment:
I wonder would ``as_edgelist`` be simpler? This would just be 
```python
def as_edgelist(self):
    edgelist = []
    for u in tree.nodes():
          for v in tree.children(u):
              edgelist.append((u, v)) 
              # or (v, u), depending on which way we think edges go. I tend to think of them
              # as going from child to parent in a tree, but this isn't very common, I think.
```

Not sure whether this representation would allow us to include branch lengths though.

Reply:
I know a dict-of-dicts-of-dicts is a bit of a mouthfull, but including branch lengths (and other annotations) will allow weighted graph traversals, which I think could be quite useful?

Reply:
Does an edge list allow orphan nodes (roots without children)? That seems to be a pretty common occurrence in the test data.

Reply:
Good points, let's stick with dict-of-dicts.